### PR TITLE
GitHub Action: Add a postgres service for testing

### PR DESCRIPTION
**Pull request by cclauss**
https://docs.github.com/en/actions/guides/creating-postgresql-service-containers

Initialize containers
* Waiting for all services to be ready
    * postgres service is healthy.
---
Run `psql --host=localhost --command='create database infobase_test;'`
```
CREATE DATABASE
```
So the GitHub Actions user `runner` can connect to postgres on localhost and create the `infobase_test` database. 👍 

But 👎 some tests that passed on Travis CI appear to fail to connect to that database...

---
test | Travis CI | GitHub Actions
--- | --- | ---
`pytest infogami` | Passes | __Fails__
`pytest tests` | Passes | Passes
`pytest test` | Passes | __Fails__
`source scripts/run_doctests.sh` | Passes | Passes

### CODE REVIEW

**Comment by hornc** [comment]
I had concerns about these tests tasks not replicating the Travis tests, and the current structure is confusing.

We definitely should not have postgres for code linting (which is the name of this task). The `| true` hide what parts of the actions are serious. I see you are trying to reduce those, which is good.  Splitting out the unit tests to their own action would help clarify which parts are working correctly. 

 I'm not certain we should be using postgres for unit tests either. The split between `test` and `tests` is unclear  it looks like `tests` is meant to be a wip replacement? -- some of the tests in `test` look like they are simulating browser access, so perhaps those are more than unit tests.  Both `test` and `tests` have their own doctest runners, looking at the github actions output, it looks like some tests are run multiple times.

For applications I favour separate unit tests rather than doctests, because we are not generating Infogami documentation at the method level.

**Comment by hornc** [comment]
Looks like many of the tests which 'pass' are skipped ---

I'm finding it hard to tell what tests actually exist because the scripts/run_doctests.sh runs more than just doctests, including the test file tests/test_doctests.py  which runs some doctests again, so things are being run multiple times and it's not clear what the correct test entry point is.

**Comment by cclauss** [comment]
Thanks for the review, Charles.  I renamed the job to match the filename and Open Library's naming convention.

I am currently not worried about any tests that run more than once.  I am instead worried about the tests that fail.

#94 --> #118 are an attempt to reorganize our tests but they have sat unreviewed for 8 months and now they can sit a bit longer until we get GitHub Actions to pass all the tests that Travis CI did.

All tests that are followed by `|| true` should eventually pass but right now @cdrini has made it clear that our priority 1 is to get GitHub Actions to pass all the tests that Travis CI did.  Other desired changes like `isort, black, mypy, pyupgrade` discussed in internetarchive/openlibrary#4776 are _blocked_ until we have __test parity between Travis CI and GitHub Actions__.

[Doctests](https://docs.python.org/3/library/doctest.html) have nothing to do with generating documentation.  They are merely tests that are written into Python docstrings.

Therefore, getting `pytest infogami` and `pytest test` to pass on GitHub Actions are the highest priority in this PR.  Any help that you can provide on that would be greatly appreciated.

**Comment by cclauss** [comment]
Arrgghh...
* This approach successfully runs Postgres at dnb=`localhost`, port'=`5432` but web.py ___requires___ that dbn=`postgres`.  
* The [GitHub Actions Postgres docs](https://docs.github.com/en/actions/guides/creating-postgresql-service-containers) show how to create a service container that runs Postgres at dnb=`localhost`, port'=`5432` but then we can use the standard GitHub Actions runner and we must apt-get install psql, etc.

Is there an easy to make `localhost` have the hostname `postgres`?

File: .github/workflows/python_tests.yml [hunk 1]
```diff
@@ -1,35 +1,51 @@
...
       - uses: actions/checkout@v2
-      - uses: harmon758/postgresql-action@v1
-        with:
-          postgresql version: '13'  # See https://hub.docker.com/_/postgres for available versions
       - uses: actions/setup-python@v2
         with:
           python-version: ${{ matrix.python-version }}
-      - run: pip install bandit black codespell flake8 isort mypy psycopg2 pytest pyupgrade
-      - run: bandit -r . || true
+      - run: pip install black codespell flake8 isort mypy pytest pyupgrade safety
```
Comment (cdrini) [inline .github/workflows/python_tests.yml:34]: We don't want safety running here; our CI should be deterministic, and not fail due to factors unrelated to our code (i.e. a dependency getting a security vulnerability). Dependabot should be catching dependency security issues, no?

```suggestion
      - run: pip install black codespell flake8 isort mypy pytest pyupgrade
```

File: .github/workflows/python_tests.yml [hunk 2]
```diff
@@ -1,35 +1,51 @@
...
+      # web.py needs to find the database on a host named postgres
+      - run: echo "127.0.0.1 postgres" | sudo tee -a /etc/hosts
+      - run: psql --host=postgres --command='create database infobase_test;'  # MUST have a --host=
+      - run: pytest infogami -s
       - run: pytest tests
-      - run: pytest test || true
-      - run: source scripts/run_doctests.sh
+      - run: pytest test
+      - run: scripts/run_doctests.sh
+      - run: safety check
```
Comment (cdrini) [inline .github/workflows/python_tests.yml:51]:
```suggestion
```

**Review by cdrini — APPROVED** [review]
Woohoo! Nice, I'm glad these are working again. Good sleuthing!

**Pull request merged**